File: src/csharp/Foundgine.Core/Execution/Mutation/MutationResult.cs
using Foundgine.Core.Abstractions;

namespace Foundgine.Core.Execution.Mutation;

public sealed record MutationResult(
    int AffectedRows,
    IReadOnlyDictionary<FieldId, object?>? ReturnedValues = null)
{
    // Compatibility name used by the PostgreSQL E2E integration tests and older
    // callers. ReturnedValues remains the canonical result representation.
    public IReadOnlyDictionary<FieldId, object?>? ReturnedFields => ReturnedValues;
}

File: src/java/foundgine-core/src/main/java/com/foundgine/core/execution/mutation/MutationResult.java
package com.foundgine.core.execution.mutation;

import com.foundgine.core.abstractions.FieldId;

import java.util.Map;

/** Port of {@code Foundgine.Core.Execution.Mutation.MutationResult}. */
public record MutationResult(int affectedRows, Map<FieldId, Object> returnedValues) {

	public MutationResult(int affectedRows) {
		this(affectedRows, null);
	}

	/**
	 * Compatibility name used by the PostgreSQL E2E integration tests and older
	 * callers. {@link #returnedValues()} remains the canonical result
	 * representation.
	 */
	public Map<FieldId, Object> returnedFields() {
		return returnedValues;
	}
}
